lib/src/signed_video_h26x_auth.c

```diff
@@ -171,8 +158,9 @@ update_link_hash_for_auth(signed_video_t *self, h26x_nalu_list_t *nalu_list)
   const size_t hash_size = self->verify_data->hash_size;
   h26x_nalu_list_item_t *item = nalu_list->first_item;
   bool found_linked_hash;
-  while (item && !found_linked_hash && !item->nalu->is_gop_sei) {
-    if (item->used_in_gop_hash && item->nalu->is_last_nalu_part) {
+  while (item && !found_linked_hash) {
+    if (item->nalu->is_last_nalu_part && item->used_in_gop_hash &&
+        item->nalu->is_first_nalu_in_gop) {
```

Comment:
If the linked hash can only be updated with I-frames, don't we get out of sync if an I-frame is lost?

Reply:
Can you add another GOP at the end to the test case `remove_the_i_nalu`
Then the test should behave like this (skipping comments)
```
test_stream_t *list = create_signed_nalus("IPPIPPIPPIPPIPPI", settings[_i]);
test_stream_check_types(list, "SIPPSIPPSIPPSIPPSIPPSI");
const int remove_nalu_number = 10;
remove_item_then_check_and_free(list, remove_nalu_number, 'I');
test_stream_check_types(list, "SIPPSIPPSPPSIPPSIPPSI");
signed_video_accumulated_validation_t final_validation = {
    SV_AUTH_RESULT_NOT_OK, false, 17, 16, 1, SV_PUBKEY_VALIDATION_NOT_FEASIBLE, true, 0, 0};
struct validation_stats expected = {.valid_gops = 3,
    .invalid_gops = 3,
    .missed_nalus = 1,
    .pending_nalus = 6,
    .final_validation = &final_validation};
if (settings[_i].auth_level == SV_AUTHENTICITY_LEVEL_FRAME) {
  expected.valid_gops = 4;
  expected.invalid_gops = 2;
}
validate_nalu_list(NULL, list, expected, true);
test_stream_free(list);
```